### DISCUSSION TITLE
Is it possible to type guards' `{ cond }` in xstate 4.32.1 with `@xstate/cli`?

### DISCUSSION THREAD

**Discussion by christopher-francisco-jpmc** [statelyai/xstate#4922] [Q&A] [answered]
In this discussion https://github.com/statelyai/xstate/discussions/2116 it's stated xstate/codegen would take care of it. I imagine it's referring to `@xstate/cli`.

Given the following dummy machine (and its corresponding `*typegen.ts` file, I'm not being able to get type safety for `cond.level`. There's also no mention of `level` in the autogen. I reckon I'm either missing something, or this use case is not handled by the tool.

Is this possible to achieve?

PS: I tried to create a sandbox for this but I couldn't get it to install specifically 4.32.1 nor to run autogen tool

```ts
const machine = createMachine({
  tsTypes: {} as import("./chrisMachine.typegen").Typegen0,
  schema: {
    context: {} as {},
    events: {} as { type: 'CONTINUE', }
  },

  initial: 'awake',

  states: {
    awake: {
      on: {
        CONTINUE: [
          {
            target: "asleep",
            cond: {
              type: 'isTiredEnough',
              level: 8
            }
          }
        ]
      }
    },

    asleep: {
      type: "final"
    }
  },
}, {
    guards: {
      isTiredEnough: (context, event, { cond }) => {
        return cond.level >= 9;
      }
    },
});
```

```ts
// This file was automatically generated. Edits will be overwritten

export interface Typegen0 {
  "@@xstate/typegen": true;
  internalEvents: {
    "xstate.init": { type: "xstate.init" };
  };
  invokeSrcNameMap: {};
  missingImplementations: {
    actions: never;
    delays: never;
    guards: never;
    services: never;
  };
  eventsCausingActions: {};
  eventsCausingDelays: {};
  eventsCausingGuards: {
    isTiredEnough: "CONTINUE";
  };
  eventsCausingServices: {};
  matchesStates: "asleep" | "awake";
  tags: never;
}
```

**Comment by davidkpiano** [comment, marked as answer]
This is likely a missing feature in Typegen. Is it possible for you to migrate to XState v5 instead?

**Reply by christopher-francisco-jpmc** [reply]
We definitely will. But we sort of need to uplift v4 before increasing the scope any further. Thanks for the answer!
